contracts/libraries/ERC721Booking.sol

```diff
@@ -30,6 +30,60 @@ abstract contract ERC721Booking is Context, ERC165, IERC721, IERC721Metadata, Re
         return bytes(baseURI).length > 0 ? string(abi.encodePacked(baseURI, tokenId.toString())) : "";
     }
 
+    /*============================================================
+                        BOOKING DATA STORAGE
+    ============================================================*/
+    struct Booking {
+        uint256 checkIn;
+        uint256 checkOut;
+        bytes data;
+    }
+
+    uint256 public bookingCounter;
+
+    mapping(uint256 tokenId => uint256) public bookingIds;
+
+    mapping(uint256 bookingId => Booking) public bookings;
+
+    function _createBooking(uint256 fromId, uint256 toId, bytes memory data) internal virtual {
+        unchecked {
+            bookingCounter++;
+        }
+
+        bookings[bookingCounter] = Booking(fromId, toId, data);
+        if (fromId == toId) {
+            bookingIds[fromId] = bookingCounter;
+        } else {
+            uint256 tokenId = fromId;
+            while (tokenId <= toId) {
+                bookingIds[tokenId] = bookingCounter;
+                unchecked {
+                    tokenId++;
+                }
+            }
+        }
+    }
+
+    function _deleteBooking(uint256 fromId, uint256 toId) internal virtual {
+        if (fromId == toId) {
+            delete bookingIds[fromId];
+        } else {
+            if (bookingIds[fromId] != bookingIds[toId]) {
+                revert MismatchedBookingIds();
+            }
+
+            uint256 tokenId = fromId;
+            while (tokenId <= toId) {
+                delete bookingIds[tokenId];
+                unchecked {
+                    tokenId++;
+                }
+            }
+        }
+
+        delete bookings[fromId];
```

Comment:
There seems to be a bug here, because `bookings` should take a booking's id as index instead of the token id.